parsl/channels/local/local.py

```diff
@@ -64,12 +65,16 @@ def execute_wait(self, cmd, walltime=None, envs={}):
                 shell=True,
                 preexec_fn=os.setpgrp
             )
+            logger.debug("Created process with pid %s. Performing communicate", proc.pid)
             (stdout, stderr) = proc.communicate(timeout=walltime)
             retcode = proc.returncode
+            logger.debug("Process returned %s", proc.returncode)
 
-        except Exception as e:
-            logger.warning("Execution of command '{}' failed due to \n{}".format(cmd, e))
+        except Exception:
+            logger.exception(f"Execution of command '{cmd}' failed")
             raise
+        else:
+            logger.debug("Execution of command completed normally")
```

Comment:
In a multiple-concurrent-commands scenario, is this enough information?  Would adding the command that completed normally to the log line be of use?  Something like:
```python
logger.debug("Execution of command completed normally: (%s) %s", retcode, cmd)
```
I don't have enough context here to know if that would just be noise, but thinking in terms of a retrospective and "how do I correlate which subprocess finished at what time?" and so forth.

Reply:
PR #2328 was intended to address pretty much exactly this kind of ambiguity by adding a bunch more context to every log line to give that kind of context. Adding the process ID of the Popened process to all of these log lines would maybe make things more explicit.

Reply:
pid added to some log lines in 9745d88